sound/soc/sof/topology.c

```diff
@@ -2183,7 +2183,13 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	process->comp.hdr.size = ipc_size;
 	process->comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
 	process->comp.id = swidget->comp_id;
-	process->comp.type = type;
+	if (sdev->fw_ready.version.abi_version >= SOF_ABI_VER(3, 17, 0)) {
+		/* use generic process component if supported */
+		process->comp.type = SOF_COMP_PROCESS;
+		process->comp.subtype = type;
```

Comment:
how is this subtype defined? and how does this help make the difference between e.g. and encoder and an EQ? You are missing one layer of description IMHO.

Reply:
also how does this relate to PR #2090? Please don't provide disjoint PRs for the same topic, it's not nice for reviewers.

Reply:
> how is this subtype defined? and how does this help make the difference between e.g. an encoder and an EQ? You are missing one layer of description IMHO.

At the moment I just use the component type as the subtype for the process components, as the EQ already has its component type. 

Thanks for your feedback Pierre. We are actually not quite aligned about what direction should we go (see the discussion here https://github.com/thesofproject/sof/pull/2956#issuecomment-631833396).

You are right that once we have decided to go with subtype, I need to add definition for the subtype (e.g. make the subtype another enum, move all existed process component types' definition into this enum, and this new defined enum should be kept in the header file shared by FW and topology only, the linux driver won't know about this).

Reply:
> also how does this relate to PR #2090? Please don't provide disjoint PRs for the same topic, it's not nice for reviewers.

Sorry for the confusion, yes, these 2 PRs are disjoint, they are 2 different solutions as described here: https://github.com/thesofproject/sof/pull/2956#issuecomment-631833396

Reply:
Man, you are really not making it easy for reviewers to look into your work...Please try and put yourself in the shoes of someone not familiar with your ideas and without context on another email thread.

As shown with PR #2138 I had to do the integration work myself to figure out what to do with your patches and what you meant in commit messages. Try and explain more what you are trying to fix, it'll help have your patches merged quicker and with fewer review cycles.

Reply:
Thanks for your suggestion here, I would forgive myself here, I joined the related discussion meetings, created 2 discussion topics, verified and provided 2 solutions for it, maybe the commit messages are still not good, but I have tried my best.

Reply:
I'll also chime in that this has been exceptionally confusing case, but for me @keyonjie 's PR have actually helper to clarify the problem. It seems the SOF discussions pages (like https://github.com/orgs/thesofproject/teams/sof-developers/discussions/32 ) work fairly well to brainstorm ideas, but we don't really seem to get actual decisions done there. To force a decision, submitting a PR works better (like @keyonjie you did here), but when you have issues that affect both driver and FW arch, this approach does not seem to work very well as the discussion threads get distributed across multiple PRs.

Not sure how to improve this. One option is drive to conclusion with a patch to sof-docs that documents the proposed design.

Reply:
Thank you @kv2019i for the constructive suggestion, let's try to finalize the solution in the next tech sync meeting, and then I will close PRs for one of the two solutions.

Reply:
having an sof-docs PR is really the only way to close on this.